portsyncd/linksync.cpp

```diff
@@ -44,19 +42,45 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
     bool oper_state = rtnl_link_get_flags(link) & IFF_LOWER_UP;
     unsigned int mtu = rtnl_link_get_mtu(link);
 
-    std::vector<FieldValueTuple> fvVector;
+    vector<FieldValueTuple> fvVector;
     FieldValueTuple a("admin_status", admin_state ? "up" : "down");
     FieldValueTuple o("oper_status", oper_state ? "up" : "down");
     FieldValueTuple m("mtu", to_string(mtu));
     fvVector.push_back(a);
     fvVector.push_back(o);
     fvVector.push_back(m);
 
+    vector<FieldValueTuple> temp;
     if (m_lagTableConsumer.get(key, temp))
         m_lagTableProducer.set(key, fvVector);
     else if (m_vlanTableConsumer.get(key, temp))
         m_vlanTableProducer.set(key, fvVector);
     else if (m_portTableConsumer.get(key, temp))
-        m_portTableProducer.set(key, fvVector);
+    {
+        /*
+         * If the port entry is just initialized without having admin_status/oper_status field,
+         * we manually bring up the port using ifup --force command.
+         */
+        bool init = true;
+        for (auto it = temp.begin(); it != temp.end(); it++)
+        {
+            if (fvField(*it) == "admin_status" || fvField(*it) == "oper_status")
```

Comment:
why check oper_status? the port orch may set the oper_status, but the linux created the host interface.